src/PowerShellEditorServices/Language/FindDeclarationVisitor.cs

```diff
@@ -76,19 +78,68 @@ public override AstVisitAction VisitFunctionDefinition(FunctionDefinitionAst fun
         /// or a decision to continue if it wasn't found</returns>
         public override AstVisitAction VisitAssignmentStatement(AssignmentStatementAst assignmentStatementAst)
         {
-            var variableExprAst = assignmentStatementAst.Left as VariableExpressionAst;
-            if (variableExprAst == null ||
-                variableName == null ||
-                !variableExprAst.VariablePath.UserPath.Equals(
+            if (variableName == null)
+            {
+                return AstVisitAction.Continue;
+            }
+
+            // The AssignmentStatementAst could contain either of the following Ast types:
+            // VariableExpressionAst, ArrayLiteralAst, ConvertExpressionAst, AttributedExpressionAst
+            // We might need to recurse down the tree to find the VariableExpressionAst we're looking for
+            List<VariableExpressionAst> asts = FindMatchingAsts(assignmentStatementAst.Left, variableName);
+            if (asts.Count > 0)
+            {
+                FoundDeclaration = new SymbolReference(SymbolType.Variable, asts.FirstOrDefault().Extent);
+                return AstVisitAction.StopVisit;
+            }
+            return AstVisitAction.Continue;
+        }
+
+        /// <summary>
+        /// Takes in an ExpressionAst and recurses until it finds all VariableExpressionAst and returns the list of them.
+        /// We expect this ExpressionAst to be either: VariableExpressionAst, ArrayLiteralAst, ConvertExpressionAst, AttributedExpressionAst
+        /// </summary>
+        /// <param name="ast">An ExpressionAst</param>
+        /// <param name="variableName">The name of the variable we are trying to find</param>
+        /// <returns>A list of ExpressionAsts that match the variable name provided</returns>
+        private static List<VariableExpressionAst> FindMatchingAsts (ExpressionAst ast, string variableName) {
+
+            // VaraibleExpressionAst case - aka base case. This will return a list with the variableExpressionAst in it or an empty list if it's not the right one.
+            var variableExprAst = ast as VariableExpressionAst;
+            if (variableExprAst != null)
+            {
+                if (variableExprAst.VariablePath.UserPath.Equals(
                     variableName,
                     StringComparison.OrdinalIgnoreCase))
+                {
+                    return new List<VariableExpressionAst>{ variableExprAst };
+                }
+                return new List<VariableExpressionAst>();
+            }
+
+            // VariableExpressionAsts could be an element of an ArrayLiteralAst. This adds all the elements to the call stack.
+            var arrayLiteralAst = ast as ArrayLiteralAst;
+            if (arrayLiteralAst != null)
             {
-                return AstVisitAction.Continue;
+                return (arrayLiteralAst.Elements.SelectMany(e => FindMatchingAsts(e, variableName)).ToList());
```

Comment:
This isn't super important but this method might be a lot more efficient in assignments with a lot of decoration if it used yield.

If you change the return type to `IEnumerable<VariableExpressionAst>` and change the other return statements to something like

```csharp
foreach (var foundVariableAst in FindMatchingAsts(convertExprAst.Child))
{
    yield return foundVariableAst;
}

yield break;
```

Then you could remove the variable name parameter and make this a more general utility function.  You could then do the actual test in the caller while still iterating only as much as you need to.